feat(Community Permissions): Update `Who holds` tokens dropdown design and flow according to the new design

noeliaSD opened the pull request:
Closes #9043

### What does the PR do

- It updates initial dropdown view for assets and collectibles to directly show the list of corresponding elements.
- It updates assets and collectibles panel when an item is selected, according to new design.
- It updates collectibles logic when amount is 1.
- It addes collectibles group item as part of the selectable's options.
- It updates `storybook` according to new changes.
- It creates reusable `TokenItem` component.
- It updates `CommunityPermissionsHelpers`.

### Affected areas

`Community Permissions` / `Who holds` dropdown

### Screenshot of functionality 

https://user-images.githubusercontent.com/97019400/213285400-b3f5a742-145f-461d-a277-fafd3ad069e3.mov

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 8ffa9d33 | [#1](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9184/1/) | 2023-01-18 20:24:12 | ~5 min | `tests-nim` | [:page_facing_up:`log`](https://ci.status.im/job/status-desktop/job/branches/job/tests-nim/job/PR-9184/1//consoleText) |

caybro commented on ui/app/AppLayouts/Chat/controls/community/ThumbnailsDropdownContent.qml:
```diff
@@ -23,87 +23,45 @@ StatusScrollView {
+                    text: model.name
+                    color: Theme.palette.directColor1
+                    font.pixelSize: 13
+                    clip: true
```
Why clip?

caybro commented on ui/app/AppLayouts/Chat/controls/community/TokenItem.qml:
```diff
@@ -0,0 +1,84 @@
+
+    property var key
+    property string name
+    property var shortName
```
why not `string`?

caybro commented on ui/app/AppLayouts/Chat/controls/community/TokenItem.qml:
```diff
@@ -0,0 +1,84 @@
+Control {
+    id: root
+
+    property var key
```
Is key something else than a `string`?

micieslak replied:
Isn't it possible to use `string` here?

micieslak commented on ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml:
```diff
@@ -4,4 +4,12 @@ QtObject {
+        Add, Update
+    }
+
+    enum FlowType {
```
This enum seems to be used only internally by `HoldingsDropdown` and imo should be defined there.

noeliaSD replied:
Ok, agree!

caybro commented on ui/app/AppLayouts/Chat/controls/community/TokenPanel.qml:
```diff
@@ -0,0 +1,89 @@
+        customHeight: 44
+        allowDecimals: true
+        keepHeight: true
+
```
empty line

micieslak commented on ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml:
```diff
@@ -4,4 +4,12 @@ QtObject {
         Asset, Collectible, Ens
     }
+
+    enum Mode {
```
with new enums here name of the file seems to be inappropriate

noeliaSD replied:
I don't think it is not appropiate. The enums here are defining some `HoldingTypes`, types themselfs, and then actions / modes related to holdings. I would keep it like this. WDYT?

caybro commented on ui/app/AppLayouts/Chat/helpers/CommunityPermissionsHelpers.qml:
```diff
@@ -3,37 +3,42 @@ pragma Singleton
+                if (item.key === key) {
                     return item
+                }
+                else if(item.subItems) {
```
No `else` after return

noeliaSD replied:
I don't understand it. Could you give me more details, please?! :pray: :sweat_smile:

caybro replied:
```qml
if (item.key === key) 
  return item

if (item.subItems) {
  const subitem = getTokenByKey(item.subItems, key)
  if (subitem !== null)
    return subitem
}
```

caybro commented on ui/app/AppLayouts/Chat/stores/CommunitiesStore.qml:
```diff
@@ -136,7 +136,10 @@ QtObject {
     function setHoldingsTextFormat(type, name, amount) {
         switch (type) {
             case HoldingTypes.Type.Asset:
+                return `${LocaleUtils.numberToLocaleString(amount)} ${name}`
```
need to `import StatusQ.Core 0.1` for this to work

caybro replied:
If you had rebased, this would stop working otherwise ;)

micieslak commented on ui/app/AppLayouts/Chat/controls/community/HoldingsDropdown.qml:
```diff
@@ -177,6 +154,50 @@ StatusDropdown {
             text: qsTr("Back")
         }
 
+        StatusSwitchTabBar {
```
Cool replacement for the custom `HoldingsTabs` :)

caybro reviewed (changes requested):
Minor stuff only, otherwise looks very nice!

micieslak commented on ui/app/AppLayouts/Chat/helpers/CommunityPermissionsHelpers.qml:
```diff
@@ -3,37 +3,42 @@ pragma Singleton
 
-        function getByKey(model, key) {
+    function getTokenByKey(model, key) {
+        if(model) {
```
early return could be used here:
```
if (!model)
   return
```
to limit indentation

micieslak commented on ui/app/AppLayouts/Chat/helpers/CommunityPermissionsHelpers.qml:
```diff
@@ -3,37 +3,42 @@ pragma Singleton
-            }
-        }
+    function getTokenShortNameByKey(assetsModel, key) {
+        var item = getTokenByKey(assetsModel, key)
```
`const`

micieslak commented on ui/app/AppLayouts/Chat/helpers/CommunityPermissionsHelpers.qml:
```diff
@@ -3,37 +3,42 @@ pragma Singleton
-        }
+    function getTokenShortNameByKey(assetsModel, key) {
+        var item = getTokenByKey(assetsModel, key)
+        if(item)
```
`if (item)`

micieslak reviewed (changes requested):
Great job, flow looks simpler and some stuff moved to separate files, cool!

Some minor issues:

![Screenshot from 2023-01-19 11-11-18](https://user-images.githubusercontent.com/20650004/213429773-70a31cc9-001a-4dd9-8b6a-8e421580a5d4.png)
Space between Update and Remove seems too big.

![Screenshot from 2023-01-19 11-19-15](https://user-images.githubusercontent.com/20650004/213430412-172476bf-4580-4581-b818-d938393c9116.png)
Horizontal scrolling should be disabled here.

micieslak commented:
@noeliaSD one more thing: please update related Figma links in Storybook :pray:

noeliaSD commented on ui/app/AppLayouts/Chat/helpers/CommunityPermissionsHelpers.qml:
```diff
@@ -3,37 +3,42 @@ pragma Singleton
 
-    function getAssetByKey(assetsModel, key) {
-        return d.getByKey(assetsModel, key)
+    function getTokenNameByKey(assetsModel, key) {
```
Oh yess, leftover!

noeliaSD commented on ui/app/AppLayouts/Chat/stores/CommunitiesStore.qml:
```diff
@@ -136,7 +136,10 @@ QtObject {
     function setHoldingsTextFormat(type, name, amount) {
```
I agree! Moved!

noeliaSD commented:
> @noeliaSD one more thing: please update related Figma links in Storybook 🙏

Not sure if you were referring to some specific bc `HoldingsDropdownPage` didn't include any. So, I've added them and updated figmas related to `Community Permissions` to point to the new design flow.

noeliaSD commented:
@micieslak @caybro updated code with your suggestions / comments. Please, take another look!

alexandraB99 commented on ui/app/AppLayouts/Chat/controls/community/ListDropdownContent.qml:
```diff
@@ -53,61 +53,20 @@ StatusListView {
         }
     }// End of Header
-    delegate: Rectangle {
+    delegate: TokenItem {
```
nice!

alexandraB99 commented on ui/app/AppLayouts/Chat/controls/community/TokenItem.qml:
```diff
@@ -0,0 +1,84 @@
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+
+Control {
```
NICE

alexandraB99 commented on ui/app/AppLayouts/Chat/controls/community/TokenItem.qml:
```diff
@@ -0,0 +1,84 @@
+        MouseArea {
+            id: mouseArea
+            anchors.fill: parent
+            cursorShape: enabled ? Qt.PointingHandCursor : Qt.ArrowCursor
```
maybe root.enabled? or where is the MouseArea's `enabled` set?

noeliaSD replied:
Here it is: `property alias isSelectable: mouseArea.enabled`

micieslak replied:
`enabled` value is propagated from parents to children, so it's fully ok imo.

noeliaSD replied:
Ok. So I will opt for removing `isSelectable` property since it is redundant, and directly use the `enabled` property instead. Thanks for both of your inputs!

alexandraB99 commented on ui/app/AppLayouts/Chat/controls/community/TokenPanel.qml:
```diff
@@ -0,0 +1,88 @@
+
+        // values from design
+        readonly property int buttonsHeight: 44
+        readonly property int defaultSpacing: 16
```
I am just wondering - creating properties impacts performance, since in this case is only referenced once, can't we just set 16 directly and in line 63 refer to root.spacing instead?

noeliaSD replied:
Removed `defaultSpacing` as suggested!

alexandraB99 reviewed (commented):
massive PR @noeliaSD, thanks! I left a few comments

alexandraB99 commented on ui/app/AppLayouts/Chat/controls/community/TokenItem.qml:
```diff
@@ -0,0 +1,83 @@
+        MouseArea {
+            id: mouseArea
+            anchors.fill: parent
+            cursorShape: enabled ? Qt.PointingHandCursor : Qt.ArrowCursor
```
could you please just root.enabled? just to make clear we're not using MouseArea's enabled

alexandraB99 reviewed (approved):
+2 for the TokenItem! just left a small comment

micieslak reviewed (commented):
Looks good but I have some questions (+ several small style adjustments)

noeliaSD commented on ui/app/AppLayouts/Chat/controls/community/TokenPanel.qml:
```diff
@@ -0,0 +1,87 @@
+        keepHeight: true
+    }
+
+    StatusButton {
```
- Is it possible? YES. We could add, for example, a new component in `HoldingsDropdown` that contains another `Loader` that loads only the modified content. 

- Is it worth it? I'm not sure. That's why I decided this small duplication. It makes the `HoldingsDropdwon` component flow cleaner and clear. 

I'm open to other suggestions but I would prefer to not add a lot of extra complexity to this component, if we can avoid that.

micieslak replied:
Fair enough ;)

noeliaSD commented on ui/app/AppLayouts/Chat/controls/community/HoldingsDropdown.qml:
```diff
@@ -155,6 +135,9 @@ StatusDropdown {
         }
     }
 
+    width: d.defaultWidth
```
Ah yes. It is intentional. I try to use this [convention](https://doc.qt.io/qt-5/qml-codingconventions.html) as we agreed in some point.

micieslak replied:
Hmm, so shouldn't setting object's properties be done before parent objects like `d` or `statesStack`?

micieslak reviewed (approved):
:rocket:

noeliaSD commented:
> 1. OPEN Minor alignment of Asset / Collectible title
>    Figma
>    ![image](https://user-images.githubusercontent.com/50769329/214159925-e33a6076-6e6d-4745-bb79-4821d16bff6e.png)
> 
> App ![image](https://user-images.githubusercontent.com/50769329/214160050-5b447733-99eb-4181-abf0-40171644b96d.png)
> 
> 2. OPEN Default number of collectibles is 1 in app but 0 in Figma
>    ![image](https://user-images.githubusercontent.com/50769329/214160577-8414985d-01c7-4477-b055-c16715ff77c8.png)

Thanks for the feedback @glitchminer !
1. Pushed changes with correct alignement:
![Screenshot 2023-01-24 at 10 08 04](https://user-images.githubusercontent.com/97019400/214252087-23bd2e1f-f33b-42f0-aff2-30660f1199b0.png)

2. We've agreed with the designer @xAlisher to keep 1 as default value for collectibles as it was in the previous design. 

Please, review it again!

glitchminer reviewed (approved):
Tested, approved
